src/main/java/org/apache/mesos/dcos/DcosCluster.java

```diff
@@ -1,19 +1,28 @@
 package org.apache.mesos.dcos;
 
-import org.apache.http.client.fluent.Content;
-import org.apache.http.client.fluent.Request;
+import org.apache.commons.io.Charsets;
 import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
 import java.net.URI;
 import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.Optional;
 
 /**
  * Instances of this class represent DC/OS clusters.
  */
 public class DcosCluster {
-    private static final String DCOS_VERSION_PATH = "/dcos-metadata/dcos-version.json";
+    private static final Logger LOGGER = LoggerFactory.getLogger(DcosCluster.class);
+
+    /**
+     * Location of file on DC/OS cluster that can be used to extract version information.
+     */
+    public static final String DCOS_VERSION_PATH = "/opt/mesosphere/active/dcos-metadata/etc/dcos-version.json";
```

Comment:
@triclambert There are only two ways to determine DC/OS version:
1. Read `/opt/mesosphere/active/dcos-metadata/etc/dcos-version.json` on the host.
2. HTTP GET `<admin-router-url>/dcos-metadata/dcos-version.json` which serves the same file in step 1.

Both approaches are similar in terms on dependency on DC/OS. 

This PR implements this dependency carefully, and doesn't bleed failure, which still makes this safe for use in non-DC/OS environments.

Thoughts ?